h5py/tests/test_group.py

```diff
@@ -436,37 +436,51 @@ def populate(self, g):
     def test_track_order(self):
         g = self.f.create_group('order', track_order=True)  # creation order
         self.populate(g)
-        self.assertEqual(list(g),
-                         [str(i) for i in range(100)])
+
+        ref = [str(i) for i in range(100)]
+        self.assertEqual(list(g), ref)
+        self.assertEqual(list(reversed(g)), list(reversed(ref)))
 
     def test_no_track_order(self):
         g = self.f.create_group('order', track_order=False)  # name alphanumeric
         self.populate(g)
-        self.assertEqual(list(g),
-                         sorted([str(i) for i in range(100)]))
+
+        ref = [str(i) for i in range(100)]
+        ref.sort()
```

Comment:
```suggestion
        ref = sorted([str(i) for i in range(100)])
```

`sorted()` is [documented](https://docs.python.org/3/library/functions.html#sorted) as returning a list, and I think the more functional style is clearer.